source/javascripts/services/semver-service.ts

```diff
@@ -36,21 +36,26 @@ class SemverService {
 		});
 	};
 
-	private reverseSorter = (verStr1: string, verStr2: string): number => {
+	reverseSort = (verStr1: string, verStr2: string): number => {
 		const ver1 = (verStr1 || this.MAXSEMVER).split(".");
 		const ver2 = (verStr2 || this.MAXSEMVER).split(".");
 
 		let notEqInd = 0;
-		while (ver1[notEqInd] == ver2[notEqInd] && notEqInd < Math.max(ver1.length, ver2.length)) {
+		while (ver1[notEqInd] == ver2[notEqInd] && notEqInd < Math.max(ver1.length - 1, ver2.length - 1)) {
 			notEqInd++;
 		}
 
 		// make WILDCARD always the lowest priority than everything else
-		ver1[notEqInd] = ver1[notEqInd] === this.WILDCARD ? "-1" : ver1[notEqInd];
-		ver2[notEqInd] = ver2[notEqInd] === this.WILDCARD ? "-1" : ver2[notEqInd];
+		ver1[notEqInd] = ver1[notEqInd] === this.WILDCARD ? "-1" : ver1[notEqInd] || "0";
+		ver2[notEqInd] = ver2[notEqInd] === this.WILDCARD ? "-1" : ver2[notEqInd] || "0";
 
 		// comply with compare interface
-		return Math.sign(parseInt(ver1[notEqInd]) - parseInt(ver2[notEqInd])) * -1;
+		const diff = Math.sign(parseInt(ver1[notEqInd]) - parseInt(ver2[notEqInd]));
+		return diff === 0 ? 0 : diff * -1;
```

Comment:
```suggestion
		return -diff;
```

Reply:
The reason I differentiate the 0 case, because the previous implementation could even return -0 (not to mention undefined), however, the expected outcomes for a comparison method are -1, 0, 1.

So I would leave it as is: or a small improvement could be:
```diff
return diff === 0 ? 0 : -diff;
```

What do you think?

Reply:
I'd handle the `undefined` case separately. But why is `-0` an issue, if `0 === -0`?

I meant the commit related to `semverService.spec.js`, as I see there's a lot of update that's only whitespace and indentation update.

Reply:
That one was VSCode entirely, but I keep an eye on it next time ;)

Reply:
Pushed the requested change ;)